File: HRMS.API/Models/Employee.cs
namespace HRMS.API.Models
{
    public class Employee
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string Position { get; set; }
        public string Department { get; set; }
        public string AccountNumber { get; set; }
        public string EmploymentStatus { get; set; } // Active, Inactive
        public DateTime JoiningDate { get; set; }
        public Salary Salary { get; set; }
    }
}

File: hrms-frontend/src/app/services/employee.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

export interface Employee {
  id: number;
  name: string;
  email: string;
  phone: string;
  position: string;
  department: string;
  accountNumber: string;
  employmentStatus: string;
  joiningDate: string;
}

@Injectable({ providedIn: 'root' })
export class EmployeeService {
  private apiUrl = 'http://localhost:5011/api/employee';
  constructor(private http: HttpClient) {}

  getAll(search?: string) {
    return this.http.get<Employee[]>(`${this.apiUrl}?search=${search || ''}`);
  }
  getById(id: number) { return this.http.get<Employee>(`${this.apiUrl}/${id}`); }
  create(emp: Employee) { return this.http.post(this.apiUrl, emp); }
  update(id: number, emp: Employee) { return this.http.put(`${this.apiUrl}/${id}`, emp); }
  delete(id: number) { return this.http.delete(`${this.apiUrl}/${id}`); }
}